docs/configuration.md

```diff
@@ -1969,6 +1987,35 @@ method3(
 )
 ```
 
+## Scala3 rewrites
+
+This section describes rules which are applied if the appropriate dialect (e.g.,
+`runner.dialect = scala3`) is selected.
+
+> This logic is not triggered via the `rewrite.rules` parameter, but by setting
+> parameters under `rewrite.scala3` subsection.
+
+### `rewrite.scala3.convertToNewSyntax`
+
+If this flag is enabled, the following new syntax will be applied:
+
+- [control syntax](https://dotty.epfl.ch/docs/reference/other-new-features/control-syntax.html)
+- [vararg splices](https://dotty.epfl.ch/docs/reference/changed-features/vararg-splices.html)
+- [imports](https://dotty.epfl.ch/docs/reference/changed-features/imports.html)
+
+### `rewrite.scala3.removeOptionalBraces`
+
+If this flag is enabled,
+[optional braces](https://dotty.epfl.ch/docs/reference/other-new-features/indentation.html)
+will be removed and significant indentation applied.
+
+The flag takes the following values:
+
+- `never`: disabled
+- `always`: enabled
+- `newSyntaxOnly`: applies only to expressions using new syntax (which includes the case when
```

Comment:
I would change the conditions around to make it harder for people to use optional braces with the old syntax.

```suggestion
- `appliesToOldSyntax`: applies also expressions using deprecated syntax 
```